lib/nav.py

```diff
@@ -36,3 +56,116 @@ def force_dir(dir, n=1):
 			if ACTIONS[OPS.MOVE][dir]():
 				break
 			ACTIONS[OPS.DIG][dir]()
+
+# move n blocks
+def dir(dir, n=1):
+	for i in range(n):
+		fuel.fuel()
+		if not ACTIONS[OPS.MOVE][dir]():
+			return False
+	return True
+
+# turn a by a given direction
+def turn(turn_dir):
+	ACTIONS[OPS.TURN][turn_dir]()
+
+class Navigator:
+	def __init__(self, location=[0,0,0], direction=CARDINALS.NORTH):
+		self.location=location
+		self.direction=direction
+	### GETTERS
+	def get_location(self):
+		return self.location.copy()
+	def get_direction(self):
+		return CARDINALS(self.direction)
+
+	### SETTERS
+	# change location depending on given direction
+	def set_location(self, dir):
+		if dir == DIRS.UP:
+			axis = 2
+			direction = 1
+		elif dir == DIRS.DOWN:
+			axis = 2
+			direction = -1
+		else:
+			axis = 0 if self.get_direction() % 2 else 1
+			direction = - 1 if self.get_direction() > 1 else 1
+		self.location[axis]+=direction
+
+	# change direction given a turn direction
+	def set_direction(self, turn_dir):
+		turn_dir = -1 if turn_dir == TURNS.LEFT else 1
+		self.direction = CARDINALS((self.direction + turn_dir) % 4)
+
+	## MOVERS
+	# move in the given direction
+	def dir(self, move_dir, n=1, to_do=None):
+		for i in range(n):
+			if to_do:
+				to_do[0](to_do[1])
```

Comment:
This to_do callback could be more flexible pretty easily

`to_do[0](*todo[1:])`, this would take a variable number of args.

(as a sidenote, I'd prefer calling it 'callback' or 'func' or something more clear than to_do)

Reply:
Also some consistency with its name in other functions as well

Reply:
done